src/Adapter/Customer/Group/Repository/GroupRepository.php

```diff
@@ -49,4 +72,16 @@ public function assertGroupExists(GroupId $groupId): void
             GroupNotFoundException::class
         );
     }
+
+    /**
+     * @param CustomerGroup $customerGroup
+     *
+     * @throws CoreException
+     *
+     * @return GroupId
+     */
+    public function create(CustomerGroup $customerGroup): GroupId
```

Comment:
```suggestion
    public function create(CustomerGroup $customerGroup, array $shopIds): GroupId
```

Reply:
I get your idea but there are a few arguments that make me think it ould be better to be explicit here:
- the `id_shop_list` is not a regular Entity field that is part of the domain, it's an internal special field from the ObjectModel and most people don't even know it exists, therefore except in the handler developed by the core chances are nobody will ever fill this property
- when this property is not set the ObjectModel automatically fallback to the automatic magic based on Context shop, which is a behaviour we aim to completely remove, that's why in recent commands we try to force explicit values on all part related to multishop
- about your question we could indeed add a check in validation because it's impossible to have an empty array, it's equivalent to no shop associated which is the same as deleting the object, so there must be at least one shop in the list, and if someone wants to create the entity for the current context then he should explicitly indicate it not rely on the magic fallback

About the validation actually, it could even be checked in the command itself